### PR TITLE
Feature/xaptc 33 Godog Cucumber Testing

### PR DESCRIPTION
**Pull request by frankquinn89**
Godog Cucumber testing

### CODE REVIEW

File: uiServices/src/services.go [hunk 1]
```diff
@@ -70,10 +77,11 @@ func validateJsonWithSchema(config []byte) bool {
 	goPath := os.Getenv("GOPATH")
 	schemaLoader := gojsonschema.NewReferenceLoader("file:///" + goPath + "/src/github.com/xtracdev/automated-perf-test/schema.json")
 	documentLoader := gojsonschema.NewBytesLoader(config)
-	logrus.Info(schemaLoader)
+	//logrus.Info(schemaLoader)
```
Comment (jenny-flynn) [inline uiServices/src/services.go]: remove commented out code

File: uiServices/test/configFile.feature [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+       "rampDelay": 15
+      }
+      """
+    And the header configsDirPath is "/uiServices/test/GodogConfig.xml"
+    Then the response code should be 201
+    And the response body should be empty
+    And the config file was created at location defined by configsPathDir with parameters:
+          """
+      {
+       "apiName": "GodogConfig",
```
Comment (jenny-flynn) [inline uiServices/test/configFile.feature]: no need to have this step which is a copy of the one in the request payload. store the request payload string in the Feature and use the 

> And the config file was created at location defined by configsPathDir

 step to compare against it.

File: uiServices/test/createConfigFile_test.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	"github.com/Sirupsen/logrus"
+	"github.com/xtracdev/automated-perf-test/uiServices/src"
+	"encoding/xml"
+	"github.com/xtracdev/automated-perf-test/perfTestUtils"
+	"encoding/json"
+	"reflect"
+)
+
+type apiFeature struct {
+	resp   *http.Response
```
Comment (jenny-flynn) [inline uiServices/test/createConfigFile_test.go]: add a `requestBody string` in the struct that you can use to store and read the request payload

File: uiServices/test/createConfigFile_test.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	if len(bodyBytes) > 0 {
+		return errors.New("Body should be empty")
+	}
+	return nil
+}
+
+func theHeaderConfigsDirPathIs(path string) error{
+	expectedConfigsPathDir :=os.Getenv("GOPATH") + "/src/github.com/xtracdev/automated-perf-test/uiServices/test/GodogConfig.xml"
```
Comment (jenny-flynn) [inline uiServices/test/createConfigFile_test.go]: for now this is fine, but we should add a requestHeaders map to the Feature and read from it to see what the header value is. Anywhere we reference `/uiServices/test/GodogConfig.xml` we should just read the request headers from `a *Feature`

File: uiServices/test/createConfigFile_test.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+	}
+
+	if
+	!reflect.DeepEqual(&expectedConfig,&actualConfig) ||
+	actualConfig.APIName != expectedConfig.APIName ||
+	actualConfig.AllowablePeakMemoryVariance != expectedConfig.AllowablePeakMemoryVariance ||
+	actualConfig.TargetHost != expectedConfig.TargetHost ||
+	actualConfig.TargetPort != expectedConfig.TargetPort ||
+	actualConfig.MemoryEndpoint != expectedConfig.MemoryEndpoint ||
+	actualConfig.AllowablePeakMemoryVariance != expectedConfig.AllowablePeakMemoryVariance ||
```
Comment (jenny-flynn) [inline uiServices/test/createConfigFile_test.go]: duplication

Note: run `go vet ./...` on the project and it will tell you if there are any issues.

**Comment by jenny-flynn** [comment]
Let's use https://github.com/stretchr/testify for assertions in our unit tests & system tests. We can make this a separate story, but it will simplify the tests a lot.

**Comment by jenny-flynn** [comment]
Merge this branch before the ui ones.

**Comment by jenny-flynn** [comment]
Also, you guys need to be mindful of code coverage. It's down to 65% now. Regularly run coverage reports to make sure you always add a test before adding functionality.

> github.com/xtracdev/automated-perf-test/uiServices/src/homepage.go	 GetIndexPage		 100.00% (4/4)
> github.com/xtracdev/automated-perf-test/uiServices/src/homepage.go	 routeConfigs		 100.00% (4/4)
> github.com/xtracdev/automated-perf-test/uiServices/src/services.go	 FilePathExist		 100.00% (2/2)
> github.com/xtracdev/automated-perf-test/uiServices/src/services.go	 ConfigCtx		 100.00% (1/1)
> github.com/xtracdev/automated-perf-test/uiServices/src/xmlWriter.go	 writerXml		 84.62% (11/13)
> github.com/xtracdev/automated-perf-test/uiServices/src/services.go	 postConfigs		 70.37% (19/27)
> github.com/xtracdev/automated-perf-test/uiServices/src/services.go	 validateJsonWithSchema	 62.50% (10/16)
> github.com/xtracdev/automated-perf-test/uiServices/src/homepage.go	 @36:18			 53.85% (7/13)
> github.com/xtracdev/automated-perf-test/uiServices/src/homepage.go	 GetRouter		 0.00% (0/6)
> github.com/xtracdev/automated-perf-test/uiServices/src/homepage.go	 StartUiMode		 0.00% (0/1)
> github.com/xtracdev/automated-perf-test/uiServices/src/services.go	 @15:26			 0.00% (0/1)
> github.com/xtracdev/automated-perf-test/uiServices/src			 ----------------------	 65.91% (58/88)

**Comment by frankquinn89** [comment]
Refactored

**Pull request closed without merging**